### ISSUE TITLE
Google AI Gemini: implement Spring Boot starter

### ISSUE THREAD

**Issue by langchain4j** [langchain4j/langchain4j#2103]
Implement Spring Boot starter for `langchain4j-google-ai-gemini` in https://github.com/langchain4j/langchain4j-spring repo

**Comment by Suhas-Koheda** [comment]
@langchain4j hey can i try working on it ?
I have just started reading docs related to gemini api in langchain4j 
I can refer to the other starter packs and create it

**Comment by langchain4j** [comment]
@Suhas-Koheda sure, thank you in advance!

**Comment by Suhas-Koheda** [comment]
@langchain4j hey is there a streaming model for google gemini ai ?
i am unable to find any class of streaming model

**Comment by langchain4j** [comment]
@Suhas-Koheda yes, https://github.com/langchain4j/langchain4j/blob/main/langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/GoogleAiGeminiStreamingChatModel.java

**Comment by Suhas-Koheda** [comment]
Also where can i contribute for some changes in docs since i found some errors
beacuse in the docs i found this  line 
 .responseFormat(ResponseFormat.JSON)  in the model builder 
i wanted it to replace it with the builder for ResponseFormat class as i was getting error
changed line: 
.responseFormat(ResponseFormat.builder().type(ResponseFormatType.JSON).build())

**Comment by Suhas-Koheda** [comment]
Okay jus found out that i was on the older version of langchain4j lol

**Comment by Suhas-Koheda** [comment]
@langchain4j  hey i have opened a pr regarding this :)

**Comment by langchain4j** [comment]
@Suhas-Koheda thanks a lot, I will try to review it ASAP

**Comment by emredmrcan** [comment]
@langchain4j  I’ve created a PR addressing the issue. It should resolve and close this issue. Please let me know if any changes are needed!

**Comment by Suhas-Koheda** [comment]
@emredmrcan excuse me hey! 
It was previously assigned to me :-) 
Thank you!

**Comment by langchain4j** [comment]
@emredmrcan thank you, but @Suhas-Koheda is already working on this issue. What's the point to spend time on the duplicated effort?

**Comment by emredmrcan** [comment]
> @emredmrcan excuse me hey! It was previously assigned to me :-) Thank you!

@Suhas-Koheda Yes, you are right. My bad! I thought you changed your direction after I just saw [this message](https://github.com/langchain4j/langchain4j/pull/2217#issuecomment-2509196536) and read the title of your PR here (Change the safetySetting builder to support map of harmCategory and HarmBlockThreshold and toolConfig to support geminiMode and ListofFunctionNames). I needed to check further. I hope my PR request will be helpful for your implementation. I am sure it is missing some test cases and there are ways to improve it. Sorry again!


> @emredmrcan thank you, but @Suhas-Koheda is already working on this issue. What's the point to spend time on the duplicated effort?

@langchain4j I added Chroma as a spring-boot-starter yesterday and I was thinking to add this quickly since I got familiar with. However, I missed the effort of @Suhas-Koheda. And you are right, there is no point to spend time on duplicated effort
